api/avatars.py

```diff
@@ -3,16 +3,202 @@
 from peewee import IntegrityError
 from werkzeug.utils import secure_filename
 from api import *
+from urllib.parse import urlparse
+
+import faf.db as db
 
 from flask import request
 
-@app.route("/avatar", methods=['GET', 'POST'])
+import logging
+logger = logging.getLogger(__name__)
+
+class Avatar:
+    """
+    Avatar model/db class
+
+    The implementation of this class is a little tricky since we have to handle
+    the url and filename.
+    """
+
+    URLBASE = app.config.get('AVATAR_URL', 'http://content.faforever.com/faf/avatars/')
+    FILEBASE = app.config.get('AVATAR_FOLDER', '/content/faf/avatars/')
+
+    def __init__(self, id=None, filename=None, url=None, tooltip=None, **kwargs):
+        """
+        Constructor
+
+        :param int id: database id
+        :param string filename: filename / url / whatever
+        :param string tooltip: tooltip
+        """
+        self.id = id
+        # FIXME: Dirty hack, done dirt cheap
+        filename = filename or url
+        if '/' in filename:
+            filename = filename.split('/')[-1]
```

Comment:
> Acccording to stackoverflow os.path.basename seems to work well with urls.

That sounds like an accidental feature likely to break any random time, and then someone will look at the code and go "why did that moron use os.path for a url". So I'm not going to change that.